src/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -70,29 +97,91 @@ export default function ColumnHeader({ columnIndex, columnName, dataReady, direc
     }
   }, [sortable, columnName, direction, orderByIndex])
 
+  const handleContextMenu = useCallback((e: MouseEvent) => {
+    e.preventDefault()
+    setMenuPosition({
+      x: e.clientX,
+      y: e.clientY,
+    })
+    setShowMenu(true)
+  }, [])
+
+  const closeMenu = useCallback(() => {
+    setShowMenu(false)
+  }, [])
+
+  // Close menu when clicking outside
+  useEffect(() => {
+    if (showMenu) {
+      const handleClickOutside = () => {
+        setShowMenu(false)
+      }
+
+      document.addEventListener('click', handleClickOutside)
+      return () => {
+        document.removeEventListener('click', handleClickOutside)
+      }
+    }
+  }, [showMenu])
+
+  // Handle menu button click
+  const handleMenuButtonClick = useCallback((e: MouseEvent) => {
+    const rect = ref.current?.getBoundingClientRect()
+    if (rect) {
+      setMenuPosition({
+        x: e.clientX,
+        y: rect.bottom,
+      })
+      setShowMenu(true)
+    }
+  }, [])
+
   return (
-    <th
-      ref={ref}
-      scope="col"
-      role="columnheader"
-      aria-sort={direction ?? (sortable ? 'none' : undefined)}
-      data-order-by-index={orderBySize !== undefined ? orderByIndex : undefined}
-      data-order-by-size={orderBySize}
-      aria-description={description}
-      // 1-based index, +1 for the row header
-      // TODO(SL): don't hardcode it, but get it from the table context
-      aria-colindex={columnIndex + 2}
-      title={description}
-      onClick={onClick}
-      style={columnStyle}
-      className={className}
-    >
-      {children}
-      <ColumnResizer
-        setWidth={setWidth}
-        onDoubleClick={autoResize}
-        width={width}
+    <>
+      <th
+        ref={ref}
+        scope="col"
+        role="columnheader"
+        aria-sort={direction ?? (sortable ? 'none' : undefined)}
+        data-order-by-index={orderBySize !== undefined ? orderByIndex : undefined}
+        data-order-by-size={orderBySize}
+        aria-description={description}
+        // 1-based index, +1 for the row header
+        // TODO(SL): don't hardcode it, but get it from the table context
+        aria-colindex={columnIndex + 2}
+        title={description}
+        onClick={onClick}
+        onContextMenu={handleContextMenu}
+        style={{ ...columnStyle, position: 'relative' }}
+        className={className}
+        aria-label={title || columnName}
+      >
+        <span className={styles.headerText}>
+          {children}
+        </span>
+        <ColumnMenuButton onClick={handleMenuButtonClick} />
+        <ColumnResizer setWidth={setWidth} onDoubleClick={autoResize} width={width} />
+      </th>
+      <ColumnMenu
```

Comment:
Is there a reason the ColumnMenu is a sibling of the `<th>`? Why not inside the `<th>`?

I'm not necessarily opposed but doesn't this result in table that looks like:

```html
<table>
  <thead>
    <th>...</th>
    <ColumnMenu/>
    <th>...</th>
    <ColumnMenu/>
...
```

Reply:
It's the `createPortal` implementation that we have in ColumnMenu. So the Column menu is attached to document.body instead, this is to prevent z-index issues.

I'll move this out into a function to render here instead and I'll add a comment.